operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentOperator.java

```diff
@@ -84,4 +84,24 @@ protected Integer currentScale(String namespace, String name) {
         Annotations.annotations(desired).put(Annotations.ANNO_DEP_KUBE_IO_REVISION, k8sRev);
         return super.internalPatch(namespace, name, current, desired, cascading);
     }
+
+    public Future<Void> generation(String namespace, String name, long pollIntervalMs, long timeoutMs) {
+        return waitFor(namespace, name, pollIntervalMs, timeoutMs, this::isGenerated);
+    }
+
+    /**
+     * Check if a deployment has been generated.
+     *
+     * @param namespace The namespace.
+     * @param name The resource name.
+     * @return Whether the deployment has been generated.
+     */
+    public boolean isGenerated(String namespace, String name) {
```

Comment:
I wonder if we can find some better name. `isGenerated` can have different expectations. What about `isObserved`?

Also, does this need to be public?